src/main/c/CMakeLists.txt

```diff
@@ -60,85 +65,53 @@ function(add_seasocks_obj_library NAME)
-)
+        )
 
-if (SEASOCKS_SHARED OR CMAKE_POSITION_INDEPENDENT_CODE)
-    add_seasocks_obj_library(seasocks_pic_obj)
-    set_property(TARGET seasocks_pic_obj PROPERTY POSITION_INDEPENDENT_CODE TRUE)
-endif ()
 
-if (SEASOCKS_SHARED)
-    add_library(seasocks_so SHARED $<TARGET_OBJECTS:seasocks_pic_obj> $<TARGET_OBJECTS:embedded>)
-    add_library(Seasocks::seasocks_so ALIAS seasocks_so)
-    target_link_libraries(seasocks_so PUBLIC $<BUILD_INTERFACE:seasocks_headers> Threads::Threads)
-    if (DEFLATE_SUPPORT)
-        target_link_libraries(seasocks_so PRIVATE ZLIB::ZLIB)
-    endif ()
-    set_target_properties(seasocks_so PROPERTIES OUTPUT_NAME seasocks VERSION ${PROJECT_VERSION})
-    install(TARGETS seasocks_so EXPORT ${PROJECT_NAME}
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    )
+add_library(Seasocks::seasocks ALIAS seasocks)
+target_link_libraries(seasocks PRIVATE Threads::Threads)
+if (DEFLATE_SUPPORT)
+    target_link_libraries(seasocks PRIVATE ZLIB::ZLIB)
 endif ()
-
-if (SEASOCKS_STATIC)
-    if (CMAKE_POSITION_INDEPENDENT_CODE OR WIN32)
-        add_library(seasocks_obj ALIAS seasocks_pic_obj)
-    else ()
-        add_seasocks_obj_library(seasocks_obj)
-    endif()
-
-    add_library(seasocks STATIC $<TARGET_OBJECTS:seasocks_obj> $<TARGET_OBJECTS:embedded>)
-    add_library(Seasocks::seasocks ALIAS seasocks)
-    target_link_libraries(seasocks PUBLIC $<BUILD_INTERFACE:seasocks_headers> Threads::Threads)
-    if (DEFLATE_SUPPORT)
-        target_link_libraries(seasocks PRIVATE ZLIB::ZLIB)
-    endif ()
-    install(TARGETS seasocks EXPORT ${PROJECT_NAME}
+install(TARGETS seasocks EXPORT ${PROJECT_NAME}
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    )
-endif ()
+        )
 
 install(DIRECTORY seasocks
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-    FILES_MATCHING PATTERN "*.h"
-)
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        FILES_MATCHING PATTERN "*.h"
+        )
 
 install(FILES "${PROJECT_BINARY_DIR}/internal/Config.h"
-    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/seasocks/Internal"
-)
+        DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/seasocks/internal"
+        )
 
+# TODO: no idea why this doesn't work
+# fails with:
+# CMake Error: install(EXPORT "Seasocks" ...) includes target "seasocks" which requires target "seasocks_headers" that is not in any export set.
 # export targets for install.
-install(
-    EXPORT ${PROJECT_NAME}
-    FILE ${PROJECT_NAME}Targets.cmake
-    NAMESPACE ${PROJECT_NAME}::
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}/
-)
+#install(
+#        EXPORT ${PROJECT_NAME}
+#        FILE ${PROJECT_NAME}Targets.cmake
+#        NAMESPACE ${PROJECT_NAME}::
+#        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}/
+#)
```

Comment:
These lines can be de-commented, I believe.

Reply:
They were (I think you were looking at an older version?)

Reply:
Indeed, github says outdated. :smile: